xds/internal/xdsclient/bootstrap/bootstrap.go

```diff
@@ -338,6 +338,20 @@ func NewConfigFromContents(data []byte) (*Config, error) {
 			if err := json.Unmarshal(v, &config.ServerListenerResourceNameTemplate); err != nil {
 				return nil, fmt.Errorf("xds: json.Unmarshal(%v) for field %q failed during bootstrap: %v", string(v), k, err)
 			}
+		case "client_default_listener_resource_name_template":
+			if !env.FederationSupport {
+				return nil, fmt.Errorf("xds: bootstrap field %v is not support when Federation is disabled", k)
```

Comment:
We have so far handled unknown fields by ignoring them (and we recently added code to log them). Why do you want to return an error here instead?

Reply:
Should just log the error. Fixed.